src/cmd/build.js

```diff
@@ -132,22 +134,27 @@ export async function defaultPackageCreator(
     filter: (...args) => fileFilter.wantFile(...args),
   });
 
-  let extensionName: string = manifestData.name;
-
-  let {default_locale} = manifestData;
-  if (default_locale) {
-    default_locale = default_locale.replace(/-/g, '_');
-    const messageFile = path.join(
-      sourceDir, '_locales',
-      default_locale, 'messages.json'
-    );
-    log.debug('Manifest declared default_locale, localizing extension name');
-    extensionName = await getDefaultLocalizedName({
-      messageFile, manifestData,
-    });
+  let packageName;
+  if (filename) {
```

Comment:
The proposed implementation does not support variables as requested in #1335.

Let's undo the current change to this file, and do something like this:

```
filename = filename.replace( ... , ... );
const packageName = safeFilename(filename);
```

If you don't know exactly how the replacement should be implemented, look at the `getDefaultLocalizedName` method for an example. To start simple, you could support the variable names "name" and "version".